### PR TITLE
Update sisu routes

### PR DESCRIPTION
**Pull request by jj-stigell**
Changes to the Sisu instance routes
* Updated sisu routes to match the new API spec.
* Added new attribute `sisuInstanceId` to the `courseInstance` model, this should prevent any situations where our internal id's get mixed up with sisu instance id's. Value is unique and nullable (incase instance does not exist in sisu but we anyway want to add it into aalto-grades) 
* Added new mock data and updated tests to handle `sisuInstanceId`.
* Naming of parameters changed to more self-explanatory. `instanceId` => `sisuInstanceId`, etc.

### CODE REVIEW

**Review by no-comma — CHANGES_REQUESTED** [review]
Good changes so far, but there are a couple more things that could be done in this PR.

This doesn't completely match our spec just yet. On success the controllers should respond with
```
{
  success: true,
  data: {
    courseInstances: ...
  }
}
```
and in failures they should use the 502 status and return an error message `external API error: ${sisuStatusCode}`.

Also I think that rather than referring to "Sisu instances" it would be better to be explicit by talking about "Sisu course instances" instead, so `sisuCourseInstanceId` etc.

File: server/src/controllers/course.ts [hunk 1]
```diff
@@ -266,6 +267,8 @@ const courseInstanceAddRequestSchema: yup.AnyObjectSchema = yup.object().shape({
     .string()
     .oneOf([GradingType.PassFail, GradingType.Numerical])
     .required(),
+  sisuInstanceId: yup
+    .string(),
```
[the remark this answers is not in the document]
Comment (jj-stigell) [reply server/src/controllers/course.ts]: added `.notRequired()`, wrapped the return data to `data` object. renamed  `sisuInstance` to `sisuCourseInstance`. I would add the error message and status code after we get the custom error types and middleware ready.

[the remark this answers is not in the document]
Comment (no-comma) [reply server/src/controllers/course.ts]: Fair enough.

**Review by no-comma — APPROVED** [review]

File: server/src/controllers/course.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export interface LocalizedString {
 
 export interface CourseData {
   // course id is either number type id in grades db or undefined when representing parsed sisu data
-  id?: number | undefined,
```
Comment (Gskartwii) [inline server/src/controllers/course.ts:29]: Just to ensure I understand, this is still meant to be optional, but the `| undefined` is being removed due to redundancy?

**Review by Gskartwii — COMMENTED** [review]
Nice, just wanted to verify a minor detail before merging this.

File: server/src/controllers/course.ts (hunk 2 above)
Comment (Gskartwii) [reply server/src/controllers/course.ts:29]: Resolved, thanks @no-comma

**Review by Gskartwii — APPROVED** [review]

**Pull request merged**